docs/start.md

```diff
@@ -65,6 +65,8 @@ with build('drive', 'v3') as service:
     # ...
 ```
 
+**Note**: Under the hood, the `build()` function retrieves a discovery artifact in order to construct the service object.  If the `cache_discovery` argument of `build()` is set to `True`, the library will attempt to retrieve the discovery artifact from the legacy cache which is only supported with `oauth2client<4.0`. If the artifact is not available in the legacy cache and the `static_discovery` argument of `build()` is set to `True`, the library will attempt to retieve the discovery artifact from the static copy of the discovery artifacts that are shipped with the library. The maintainers recommend keeping `static_discovery=True`, which is the default, in order to avoid reliability issues related to transient network errors. The library currently does not detect breaking changes of discovery artifacts, and as a result, user code is not guaranteed to continue to succeed or fail with different versions of `google-api-python-client`. Users can set `static_discovery=False` as workaround if there is an incompatible discovery artifact shipped with the library. In addition, users may roll back the version of `google-api-python-client` if there is a breaking change in a discovery artifact. The maintainers of `google-api-python-client` encourage users to report issues with discovery artifacts so that the discovery artifacts can be corrected.
```

Comment:
Could we describe this more in terms of the users' perspectives?  Something like the following

If reliability is most important, then set static_discovery=True.  It will use the service definition shipped in the library.
If always using the most recent version of the library is most important, then ...

Reply:
Done. I also removed the text about breaking change detection as that bug is captured in #1154 . Please can you take another look?

Reply:
In an effort to move this PR forward, I'm going to resolve this comment as the feedback has been addressed.